nemo/src/types.rs

```diff
@@ -143,11 +143,17 @@ impl LogicalTypeEnum {
             },
             Self::Float64 => match gt {
                 Term::NumericLiteral(NumericLiteral::Double(d)) => DataValueT::Double(d),
+                Term::NumericLiteral(NumericLiteral::Decimal(a, b)) => {
+                    DataValueT::Double(Double::from_number(format!("{a}.{b}").parse().unwrap()))
+                }
+                Term::NumericLiteral(NumericLiteral::Integer(a)) => {
+                    DataValueT::Double(Double::from_number(a as f64))
+                }
                 Term::RdfLiteral(RdfLiteral::DatatypeValue {
                     ref value,
                     ref datatype,
                 }) => match datatype.as_str() {
-                    XSD_DOUBLE => DataValueT::Double(
+                    XSD_DOUBLE | XSD_DECIMAL | XSD_INTEGER => DataValueT::Double(
```

Comment:
Do we really want to implicitly convert integers to doubles, possibly losing precision?

Reply:
I have no real opinion about this. Generally I am against implicit conversions, but in this case we are talking about constants somebody types into a rules file. I could add a check that the parsed integer is smaller than 2**53 or disallow integer terms in double columns entirely, but then again, if the column type was defined to be double the user willingly allows loosing precision.

Reply:
Also there would be the possibility to allow 2 but disallow "2"^^xsd:integer. Not sure.

Reply:
> Also there would be the possibility to allow 2 but disallow "2"^^xsd:integer. Not sure.

yeah, `2` should be accepted, but if somebody bothers to write `"2"^^xsd:integer`, I would expect that they care about XSD semantics, and those clearly state that `xsd:integer` and `xsd:double` have distinct value spaces.

Reply:
Not sure how easy that would be to realise without major changes, though.

Reply:
On second thought, it feels that if someone tries to put a "2"^^xsd:integer into a float column, they probably made a mistake. So I will remove that cast for now.

Reply:
Agreed :)